airflow/utils/log/file_task_handler.py

```diff
@@ -56,16 +55,19 @@ def close(self):
         if self.handler:
             self.handler.close()
 
-    def read(self, task_instance):
+    def read(self, task_instance, try_number):
```

Comment:
Yes but we need to fetch logs given different try_number. We can move the entire logic of reading task_instnace logs inside this function too.